### ISSUE TITLE
Way to print which `Mock`s are registered to a `MockServer`

### ISSUE THREAD

**Issue by tgross35** [alexliesenfeld/httpmock#92]
It would be nice to have a way to see e.g. what URLs are matched or what the rules are. Part of this could just be a `Debug` impl on `Mock` that prints the fields of `RequestRequirements` that are `Some` in a nice way. 

Maybe `MockServer` could have a `fn (&self) -> impl Iterator<Item=&Mock>` somehow? I am less sure how those are stored internally.

**Comment by Tesel** [comment]
Doubling that. 
Having 500's recently because request doesn't match any mock in parallel test execution and it only happens on specific GH runner. Locally it passes and it would be useful to debug this. Hard to even create MRE
